implementations/rust/ockam/ockam_app/src/tcp/outlet/list.rs

```diff
@@ -1,23 +1,25 @@
-use crate::ctx::TauriCtx;
-use crate::Result;
+use crate::error::TauriCommandResult;
+
 use ockam::Context;
 use ockam_api::cli_state::traits::StateDirTrait;
-use ockam_api::nodes::models::portal::OutletList;
+use ockam_api::nodes::models::portal::OutletStatus;
 use ockam_command::node::initialize_node_if_default;
 use ockam_command::util::embedded_node;
 use ockam_command::{tcp, CommandGlobalOpts, GlobalArgs};
 
 /// List TCP outlets of the default node.
 #[tauri::command]
-pub fn list(_ctx: &TauriCtx) -> Result<OutletList> {
+pub fn tcp_outlet_list() -> TauriCommandResult<Vec<OutletStatus>> {
     let opts = CommandGlobalOpts::new(GlobalArgs::default());
     initialize_node_if_default(&opts, &None);
-    let res = embedded_node(rpc, opts)?;
-    Ok(res)
+    match embedded_node(rpc, opts) {
+        Ok(res) => Ok(res),
+        Err(e) => Err(format!("Error while listing TCP outlets: {e:?}")),
+    }
```

Comment:
Is `Result::map_err` helpful here? Are we expecting the body of the success arm to also evolve?

Reply:
> Are we expecting the body of the success arm to also evolve?
 I don't think so. I've refactored that to use the `map_err` function.